compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -529,7 +538,7 @@ class ExpressionCodegen(
         )
 
     override fun visitClass(declaration: IrClass, data: BlockInfo): PromisedValue {
-        classCodegen.generateLocalClass(declaration, irFunction.isInline).also {
+        classCodegen.generateLocalClass(declaration, generateSequence(this) { it.inlinedInto }.any { it.irFunction.isInline }).also {
```

Comment:
Please note there is also withinInline check inside generateLocalClass, maybe it's a bug with it?

Reply:
No, this fix is for cases like this:

```kotlin
inline fun f() {
    someOtherInlineFunction {
        object : { ... }
    }
}
```

Here we have a ClassCodegen for the file class, then a FunctionCodegen+ExpressionCodegen for `f`, then finally a FunctionCodegen+ExpressionCodegen for the lambda which calls `classCodegen.generateLocalClass`. There are no ClassCodegens between the two ExpressionCodegens.

Reply:
As I see it should be ClassCodegen + FunctionCodegen+ExpressionCodegen(1) for 'f' than same for lambda(2) and anonymous object(3). and withinInline flag should be passed from ExpressionCodegen(1) to ClassCodegen(2) to FC(2) to EC(2) and finally to CC(3)

Reply:
Oh seems it's cause of change and a bit new semantic of 
```val isInlineLambda: Boolean = false``` -> ```val inlinedInto: ExpressionCodegen```
will look more deeply